src/main/java/de/zalando/ep/zalenium/proxy/DockerSeleniumRemoteProxy.java

```diff
@@ -413,11 +413,15 @@ public boolean shutdownIfStale() {
             LOGGER.info(String.format("[%s] is stale.", getContainerId()));
             timeout("proxy being stuck | stale during a test.", ShutdownType.STALE);
             return true;
+        } else if (isTestSessionLimitReached() && !isBusy()) {
```

Comment:
Why does it need to be on the stale one too? Does it run more often?

Reply:
Yep. Stale runs even when the pool doesn't have more than minimum desired number. If the check isn't also done here, containers that have run more than their max number of sessions won't be removed until after a new container is spun up to service another request.

Means that the effectively useless container is kept running longer than it should, and the test has to wait for a new container to be spun up (though this shouldn't take long).

It would also mean that if mix == max, then the pool would be full of useless containers and not be able to create a new one.